### DISCUSSION TITLE
Interpolation of Hessian in new version

### DISCUSSION THREAD

**Discussion by tveskaeg** [SSCHAcode/python-sscha#97] [Q&A] [answered]
Dear developers,

we have already talked about this topic here, and I'm referring to the H3S example treated there:
https://github.com/SSCHAcode/python-sscha/discussions/49

Since there have been a few bugfixes and branch merges, I'd have some follow-up questions for the new code version (SSCHA 1.2, CC 1.1).

I have finished a calculation on H3S in the 222 supercell (SSCHA, 3rd and 4th Hessian), and I would like to interpolate the Hessians to a higher grid like 444 or 666. I managed to interpolate the SSCHA auxiliary matrices to the higher grids with the help of the harmonic support matrices and I have the real-space 3rd order file coming from the 222 calculation.
My questions now are:
- Which routine should I use to get the Hessians on the higher grid?
- Does the new version offer the possibility to interpolate the 4th order Hessians as well?

Thanks in advance.

Best wishes,
Roman

**Comment by mesonepigreco** [comment, marked as answer]
Hi Roman,
There are two ways.

1) The easiest approach is to simply interpolate the Hessian matrix you get from get_free_energy_hessian as you did for the SSCHA matrix (maybe using the interpolated SSCHA matrices as support dyn to improve the convergence).
2)  You can recompute the Hessian on any arbitrary q point. This is the best approach; you can follow the tutorial on the [Spectral functions](http://sscha.eu/Tutorials/tutorial_spectral/ ) to obtain the hessian computed directly on a q-path with the function:
CC.Spectral.get_static_correction_along_path

Method 2 is more precise and does a better job as you are interpolating the 2nd and 3rd order force constants to evaluate the hessian in each specific q point. However, as it is implemented, method 2 does not allow interpolating the 4th order force constant. So if you see differences between using get_free_energy_hessian with and without include_v4, you must use method 1.

The interpolation of the fourth-order force constants is much more complex and probably requires a brand new algorithm to compute the Hessian matrix with CC.Spectral.get_static_correction_along_path. We are working on it and will release this feature as a major update of the code, like version 2.0, but it will likely take some time.

Bests,
Lorenzo
